### PR TITLE
fix(editor): reduce asset size 

### LINKED ISSUE #696: Reduce asset size for the initial load of Editor
Currently, the Editor requires ~100MB of data to be transferred for the initial page load. The demo example image files are the ones that make the loading too slow. We will need to optimize the loading, e.g., compress image files, reduce their sizes given that we only use them for showing thumbnails, and load images only when a user clicks on the example gallery.

image

Image directory: https://github.com/gosling-lang/gosling.js/tree/master/editor/example/thumbnails

### PR DESCRIPTION
**Pull request by etowahadams**
Fix #696 
Toward:  #853, #650

## Change List
 - Resize the example thumbnails so the maximum with is 500px. The total combined size of the thumbnails is now ~10 MB
 - Pull out the editor examples into its own component (EditorExamples) and add types 
 - Make thumbnail images lazy load 

## Checklist
 - [ ] Ensure the PR works with all demos on the online editor
 - [ ] Unit tests added or updated
 - [ ] Examples added or updated
 - [ ] [Documentation](https://github.com/gosling-lang/gosling-website) updated (e.g., added API functions)
 - [ ] Screenshots for visual changes (e.g., new encoding support or UI change on Editor)

### CODE REVIEW

File: editor/editor.tsx [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
-import './editor.css';
+import { ICONS, type ICON_INFO } from './icon';
+import { getHtmlTemplate } from './html-template';
+import ErrorBoundary from './error-boundary';
+import { traverseTracksAndViews } from '../src/core/utils/spec-preprocess';
+import { examples, type Example } from './example';
+import EditorPanel, { type EditorLangauge } from './EditorPanel';
+import EditorExamples from './EditorExamples';
+
+import './Editor.css';
```
Comment (etowahadams) [inline editor/editor.tsx:28]: Reorder the imports. We should consider having a eslint rule to do this automatically

File: editor/editor.tsx [hunk 2]
```diff
@@ -1381,84 +1388,14 @@ function Editor(props: RouteComponentProps) {
...
-                </div>
-            </div>
+            {showExamples && (
+                <EditorExamples
+                    setShowExamples={setShowExamples}
+                    closeDescription={closeDescription}
+                    setIsImportDemo={setIsImportDemo}
+                    setDemo={setDemo}
+                />
+            )}
```
Comment (etowahadams) [inline editor/editor.tsx:1398]: Pull out the editor examples into its own component. The conditional rendering is much more clear this way

File: editor/EditorExamples.css [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    height: 30px;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    overflow: hidden;
+    display: block;
+}
+
+.example-card-bg:hover {
+    background-position: bottom right;
+}
```
Comment (etowahadams) [inline editor/EditorExamples.css:130]: Pulled out from Editor.css

File: editor/EditorExamples.css
Comment (etowahadams) [inline editor/EditorExamples.css:1]: Styles for editor examples were moved from Editor.css to EditorExamples.css

File: editor/editor.tsx (hunk 1 above)
Comment (sehilyi) [reply editor/editor.tsx:28]: It would be good to have!

File: editor/editor.tsx (hunk 2 above)
Comment (sehilyi) [reply editor/editor.tsx:1398]: Thanks for making this change!

**Review by sehilyi — APPROVED** [review]
Looks great!

Do you want to write instructions in `CONTRIBUTING.md` about how you compressed the images? We will be adding more examples to Editor, and having that will enable people (incl. us) to make consistent thumbnails.

**Comment by etowahadams** [comment]
I'll add instructions about compressing to CONTRIBUTING.md! I just used the built-in Preview to batch resize the images, but in theory imagemagick, ffmpeg etc could be used

File: CONTRIBUTING.md
Comment (etowahadams) [inline CONTRIBUTING.md:1]: Let me know if these instructions seem sufficient

Comment (sehilyi) [reply CONTRIBUTING.md:1]: Looks great! Thanks for adding this!

**Pull request merged**